Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs

```diff
@@ -218,8 +218,8 @@ public IContentStore CreateContentStore(AbsolutePath localCacheRoot)
                         PinConfiguration = pinConfiguration,
                         EmptyFileHashShortcutEnabled = _distributedSettings.EmptyFileHashShortcutEnabled,
                         RetryIntervalForCopies = _distributedSettings.RetryIntervalForCopies,
-                        EnableProactiveCopy = _distributedSettings.EnableProactiveCopy,
                         TimeoutForProactiveCopies = TimeSpan.FromMinutes(_distributedSettings.TimeoutForProactiveCopiesMinutes),
+                        ProactiveCopyMode = (ProactiveCopyMode)Enum.Parse(typeof(ProactiveCopyMode), _distributedSettings.ProactiveCopyMode),

```

Comment:
nit: I would prefer to create an extension/helper method for parsing the input.

Right now if the settings are not correct the error message would be: "Requested value 'yourenumvaluewithtypo' was not found.

So an alternative may be a dedicated helper that will throw a bit more descriptive error message like:

"Failed to parse ProactiveCopyMode property 'yourenumvalueiwthtypeo'.'.

But this is minor, because the default error message contains an invalid value.

Reply:
Leaving as is, which follows the normal pattern used in our code.